src/main/java/io/openio/sds/storage/rawx/RawxClient.java

```diff
@@ -293,32 +317,39 @@ public UploadResult call() {
 				} catch (IOException e1) {
 					logger.warn(e1);
 				}
-				throw e;
 			}
 		}
-		consume(data, size, gens, futures);
 
 		try {
-			ArrayList<UploadResult> successes = new ArrayList<UploadResult>();
-			for (Future<UploadResult> f : futures) {
-				UploadResult result = f.get();
+			quorumOrFail(pos, quorum, futures);
+
+			consume(data, size, gens, futures);
+		} catch (Exception e) {
+			// Cancel the tasks
+			int notTerminated = 0;
+			for (Future<UploadResult> future : futures) {
+				if (!future.isDone())
+					notTerminated++;
+				future.cancel(true);
+			}
+			String message = "Stream read error";
+			if (notTerminated > 0)
+				message += " (" + notTerminated + " upload jobs cancelled)";
```

Comment:
The odds are low, but in case the exception is raised by `quorumOrFail`, the message will tell "Stream read error", whereas the problem comes from the thread pool.